### DISCUSSION TITLE
update record where unset value

### DISCUSSION THREAD

**Discussion by boilemmashem** [prisma/prisma#22453] [Q&A] [answered]
### Question

I'm new to express and prisma, trying to update a MongoDB record.

MongoDB entry:
```
_id 658045373a921d5d8e1e30fc
email "email@example.com"
username "exampleuser"
password "somehash"
createdAt 2023-12-18T13:12:23.107+00:00
```

Prisma query
```js
await prisma.user.update({
    where: {
        email: req.body.email,
        registerToken: null // Unset value in mongo
    },
    data: {
        lastSignIn: new Date(),
    },
});
```

Explicitly setting `registerToken: null` in the mongodb record means the query will work. Is there a correct way to filter on an unset value with Mongo?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
model User {
  id            String    @id @default(auto()) @map("_id") @db.ObjectId
  email         String?   @unique
  username      String    @unique
  password      String
  createdAt     DateTime  @default(now())
  lastSignIn    DateTime?
  registerToken String?
}
```

- OS: macOS
- Database: MongoDB Atlas
- Node.js version: 20.2.0

- prisma                  : 5.7.0
@prisma/client          : 5.7.0
Computed binaryTarget   : darwin
Operating System        : darwin
Architecture            : x64
Node.js                 : v20.2.0
Query Engine (Node-API) : libquery-engine 79fb5193cf0a8fdbef536e4b4a159cad677ab1b9 (at node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Schema Engine           : schema-engine-cli 79fb5193cf0a8fdbef536e4b4a159cad677ab1b9 (at node_modules/@prisma/engines/schema-engine-darwin)
Schema Wasm             : @prisma/prisma-schema-wasm 5.7.0-41.79fb5193cf0a8fdbef536e4b4a159cad677ab1b9
Default Engines Hash    : 79fb5193cf0a8fdbef536e4b4a159cad677ab1b9
Studio                  : 0.495.0

**Comment by ludralph** [comment, marked as answer]
Hi @mikehaart 👋 

Thank you for raising this question.

To filter on an unset property, you can use the [isSet](https://www.prisma.io/docs/orm/reference/prisma-client-reference#isset) filter.

For example:
```js
await prisma.user.update({
    where: {
        email: req.body.email,
        registerToken: { isSet: false }
    },
    data: {
        lastSignIn: new Date(),
    },
});
```
If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
